src/lib/comms/include/sol-http-server.h

```diff
@@ -94,10 +95,35 @@ struct sol_http_progressive_response;
  * @param port The port where the server will bind.
  *
  * @return a handle to the server on success, otherwise @c NULL is returned.
+ *
+ * @see sol_http_server_new()
  */
 struct sol_http_server *sol_http_server_new(uint16_t port);
 
 /**
+ * @brief Creates a HTTPS server, binding on all interfaces
+ * in the specified @a port.
+ *
+ * With the returned handle it's possible to register paths using
+ * @c sol_http_server_register_handler()
+ * and dirs with @c sol_http_server_add_dir().
+ *
+ * @note Only one instance of sol_http_server is possible per port.
+ * Try to run a second instance in the
+ * same port will result in fail.
+ *
+ * @param port The port where the server will bind.
+ * @param cert The certificate that should be used to crypto the data.
+ * @param key The key of @a cert.
+ *
+ * @return a handle to the server on success, otherwise @c NULL is returned.
+ *
+ * @see sol_http_server_new()
+ */
+struct sol_http_server *sol_http_server_secure_new(uint16_t port,
```

Comment:
I like that approach for multiple reasons:
 - you don't have to set the parameters you don't need (as opposed to pass multiple NULL, 0...)
 - we can keep reduced number of symbols;
 - no need to change the symbol version on changes, only the struct API version
 - API version can be handled in a customized version, it's just us changing the `==` check to `<` and code to handle the changes. This can offer a better API stability.